contributors/design-proposals/api-machinery/customresources-validation.md

```diff
@@ -381,6 +381,35 @@ Note:
 
 2. For migration of CRDs with no validation to CRDs with validation, we can create a controller that will validate and annotate invalid CRs once the spec changes, so that the custom controller can choose to delete them (this is also essentially the status condition of the CRD). This can be achieved, but it is not part of the proposal.
 
+### Admission Webhook
+
+Custom resource definitions use the the same REST endpoint as built-in Kubernetes API objects. Consequently standard
+[dynamic web-hook admission controllers](https://kubernetes.io/docs/admin/extensible-admission-controllers/#external-admission-webhooks) 
+can be used for validation of custom resource definitions in addition to built-in resources.
+
+However, because the creation of admission controllers is a fairly high-privilege activity,
+and in many cases the creator of a `CustomResourceDefinition` is third-party extension code
+(e.g. an instance of the _operator pattern_) which should not have the ability to
+create arbitrary admission controllers, it is necessary to have an alternate solution for
```

Comment:
Seeing operators in the wild that install CRDs and RBAC rules, I question that there is a real difference to "creation of admission controllers" today. Of course, I totally agree that there should be. But for that we need a similar in-CRD mechanism for RBAC rules.

Reply:
That is clearly a job for a future design PR :)

Reply:
sure, just thinking out loud here ;)